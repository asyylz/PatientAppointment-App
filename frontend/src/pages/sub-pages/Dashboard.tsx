import React from 'react';
import classes from './Dasboard.module.css';
import { useSelector } from 'react-redux';
import GlobalLink from '../../components/UI/GlobalLink';
const Dashboard: React.FC = () => {
  const {
    entities: { token, data },
    image,
    status,
    error,
  } = useSelector((state: RootState) => state.currentUser);

  return (
    <div
      //style={{ border: '3px solid green' }}
      className={classes.container}
    >
      <div className={classes.userLeftSection}>
        <div className={classes.wrapper}>
          <div className={classes.image}>
            <img src={image} alt="" />
          </div>
          <p>Welcome {data.currentUser.name}</p>
        </div>
      </div>
      <div className={classes.userRightSection}>
        <div className={classes.rightTop}>
          <div className={classes.box}>1</div>
          <div className={classes.box}>1</div>
          <div className={classes.box}>1</div>
        </div>
        <div className={classes.rightBottom}>
          <div className={classes.appointments}>
            <div className={classes.wrapper1}>
              <h2>Latest Appointments</h2>
              <GlobalLink text="Book Now"></GlobalLink>
            </div>
            <hr />
            <div className={classes.wrapper2}>
              <div className={classes.appointmentBox}>1</div>
              <div className={classes.appointmentBox}>2</div>
              <div className={classes.appointmentBox}>3</div>
            </div>
          </div>
        </div>
      </div>
    </div>
  );
};

export default Dashboard;
