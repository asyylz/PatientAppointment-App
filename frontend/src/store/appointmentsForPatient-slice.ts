import {
  fetchEntitiesWithIdAndToken,
  createEntitySlice,
} from './create-generic-slice';


export const patientAppointmentsUrl = (id: string, pagination?: number) => {
  return `http://localhost:3000/api/v1/appointments/patients/${id}?limit=10&page=${pagination}`;
};

export const fetchAppointmentsForPatient =
  fetchEntitiesWithIdAndToken<AppointmentsForPatient>(
    'appointmentsForPatient',
    patientAppointmentsUrl
  );

const appointmentsForPatientSlice = createEntitySlice<AppointmentsForPatient>(
  'appointmentsForPatient',
  fetchAppointmentsForPatient
);

const { actions, reducer } = appointmentsForPatientSlice;

export const appointmentsForPatientActions = actions;
export default reducer;
