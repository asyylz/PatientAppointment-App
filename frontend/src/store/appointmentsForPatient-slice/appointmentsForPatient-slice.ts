import {
  fetchEntitiesWithId,
  createEntitySlice,
} from '../create-generic-slice/create-generic-slice';
const apiURL =
  import.meta.env.VITE_NODE_ENV === 'production'
    ? import.meta.env.VITE_SERVER_URL
    : import.meta.env.VITE_LOCAL_URL;
    
export const patientAppointmentsUrl = (id: string, pagination?: number) => {
  return `${apiURL}/appointments/patients/${id}?limit=10&page=${pagination}`;
};

export const fetchAppointmentsForPatient =
  fetchEntitiesWithId<AppointmentsForPatient>(
    'appointmentsForPatient',
    patientAppointmentsUrl
  );

const appointmentsForPatientSlice = createEntitySlice<AppointmentsForPatient>(
  'appointmentsForPatient',
  fetchAppointmentsForPatient
);

const { actions, reducer } = appointmentsForPatientSlice;

export const appointmentsForPatientActions = actions;
export default reducer;
