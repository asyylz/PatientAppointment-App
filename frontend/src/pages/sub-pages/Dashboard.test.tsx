import { fireEvent, render, screen, within } from '@testing-library/react';
import { Provider } from 'react-redux';
import { act } from 'react';
import renderer from 'react-test-renderer';
import store from '../../store/index';
import { MemoryRouter } from 'react-router-dom';
import '@testing-library/jest-dom';
import '@testing-library/dom';
import { fetchAppointmentsForPatient } from '../../store/appointmentsForPatient-slice';
import Dashboard from './Dashboard';
import {
  deleteAppointmentMock,
  updateAppointmentMock,
} from '../../_testUtils/mocks/mockHttp';
/* ------------------- Mock useEffect ------------------- */
jest.mock('react', () => ({
  ...jest.requireActual('react'),
  useEffect: jest.fn(),
}));

 /* ------------------- Mock useHttp ------------------ */
// it needs to be declared at the top level of the test file to override the module system before any imports occur. 
jest.mock('../../hooks/useHttp');
 
/* -------------------- Initial state ------------------- */
const initialState = {
  currentUser: {
    status: 'success',
    token:
      'eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpZCI6IjY2NmMxZjEwYWQwMTVlNmRmMTdhOThlNCIsImlhdCI6MTcyMzI0MzI0NywiZXhwIjoxNzIzMjQ1MDQ3fQ.7QOdK3nCWtb5wQ1yWZD26GhZ731F8nZzlH-d9-oonrY',
    userData: {
      _id: '6673662fbd42a966b75dec92',
      name: 'Asiye',
      email: 'alice@test.com',
      DOB: '1986-01-22T00:00:00.000Z',
      image: 'test image',
      role: 'doctor',
      address: {
        street: '123 Main St',
        city: 'New York',
        country: 'NY',
        town: 'test town',
        postalCode: '10001',
      },
    },
    _persist: { version: -1, rehydrated: true },
  },
  departments: { entities: [], status: 'idle', error: null },
  doctors: {
    entities: [],
    status: 'idle',
    error: null,
    _persist: { version: -1, rehydrated: true },
  },
  reviews: { entities: [], status: 'idle', error: null },
  search: '',
  appointmentsForDoctor: { entities: [], status: 'idle', error: null },
  appointmentsForPatient: { entities: [], status: 'idle', error: null },
};

/* ------------------- After dispatch ------------------- */
const afterDispatchState = {
  entities: {
    appointmentsForPatient: [
      {
        _id: '66b0a0c31d2124f7d8aaa65b',
        doctorId: {
          _id: '665f0f2959c971659af920ca',
          firstName: 'Bowen',
          lastName: 'Chan',
          departmentId: '6660fbd12e3c00fe18cfceef',
        },
        patientId: '6673662fbd42a966b75dec92',
        appointmentDateAndTime: '2024-08-16T11:00:00.000Z',
        reason: 'Before update',
        diagnose: null,
        status: null,
        referral: false,
      },
    ],
  },
  status: 'succeeded',
  error: null,
};

/* ---------------- Test Component ------------------ */
describe('Dasboard', () => {
  afterEach(() => {
    jest.clearAllMocks();
  });
  /* -------------------------- - ------------------------- */
  it('1--Renders correctly before data is fetched', () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <MemoryRouter>
            <Dashboard />
          </MemoryRouter>
        </Provider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
  /* -------------------------- - ------------------------- */
  it('2--Initializes state with status "idle', () => {
    render(
      <Provider store={store}>
        <MemoryRouter>
          <Dashboard />
        </MemoryRouter>
      </Provider>
    );
    // Initial state should be 'idle'
    expect(store.getState().appointmentsForPatient.status).toEqual('idle');
    expect(store.getState().appointmentsForPatient).toEqual(
      initialState.appointmentsForPatient
    );
  });
  /* -------------------------- - ------------------------- */
  it('3--Dispatches fetchAppointmentsForPatient action and updates state with the action payload', async () => {
    expect(store.getState().appointmentsForPatient.status).toEqual('idle');

    render(
      <Provider store={store}>
        <MemoryRouter>
          <Dashboard />
        </MemoryRouter>
      </Provider>
    );

    await act(async () => {
      store.dispatch(
        fetchAppointmentsForPatient({
          id: '6673662fbd42a966b75dec92', // id same with initialState currentUser Id
          token:
            'eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpZCI6IjY2NmMxZjEwYWQwMTVlNmRmMTdhOThlNCIsImlhdCI6MTcyMzI0MzI0NywiZXhwIjoxNzIzMjQ1MDQ3fQ.7QOdK3nCWtb5wQ1yWZD26GhZ731F8nZzlH-d9-oonrY',
        })
      );
    });

    expect(store.getState().appointmentsForPatient.status).toEqual('loading');
    //console.log(store.getState().appointmentsForPatient.status);
    const fullfilledAction = {
      type: 'appointmentsForPatient/fetchWithIdAndToken/fulfilled',
      payload: {
        appointmentsForPatient: [
          {
            _id: '66b0a0c31d2124f7d8aaa65b',
            doctorId: {
              _id: '665f0f2959c971659af920ca',
              firstName: 'Bowen',
              lastName: 'Chan',
              departmentId: '6660fbd12e3c00fe18cfceef',
            },
            patientId: '6673662fbd42a966b75dec92',
            appointmentDateAndTime: '2024-08-16T11:00:00.000Z',
            reason: 'Before update',
            diagnose: null,
            status: null,
            referral: false,
          },
        ],
      },
    };

    await act(async () => {
      store.dispatch(fullfilledAction);
    });
    console.log(store.getState().appointmentsForPatient.status);
    console.log(store.getState().appointmentsForPatient.entities);
  });

  /* -------------------------- - ------------------------- */
  it('4--Renders fetched data in the DOM after dispatching action', () => {
    render(
      <Provider store={store}>
        <MemoryRouter>
          <Dashboard />
        </MemoryRouter>
      </Provider>
    );

    // After dispatching the action, the state should be updated
    expect(store.getState().appointmentsForPatient).toEqual(afterDispatchState);
    // console.log(store.getState().appointmentsForPatient);
  });
  /* -------------------------- - ------------------------- */
  it('5--Renders 7 cells per row with correct values based on fetched data', () => {
    render(
      <Provider store={store}>
        <MemoryRouter>
          <Dashboard />
        </MemoryRouter>
      </Provider>
    );
    const cells = screen.getAllByRole('cell');
    expect(cells.length).toBe(7);
    // Check the content of the cells
    expect(cells[0]).toHaveTextContent('1.');
    expect(cells[1]).toHaveTextContent('Dr. Bowen Chan');
    expect(cells[2]).toHaveTextContent('16/08/2024 Friday');
    expect(cells[3]).toHaveTextContent('11:00');
    expect(cells[4]).toHaveTextContent('Before update');
    expect(cells[5]).toHaveTextContent('');
    expect(cells[6]).toContainHTML('svg');
  });
  /* -------------------------- - ------------------------- */
  it('6--Renders the correct number of rows based on the fetched data length', () => {
    render(
      <Provider store={store}>
        <MemoryRouter>
          <Dashboard />
        </MemoryRouter>
      </Provider>
    );
    const rows = within(
      screen.getByTestId('appointments-patient')
    ).getAllByRole('row');
    // we have two <tr> tag for decoration,  they need to be taken out
    expect(rows.length - 2).toEqual(
      afterDispatchState.entities.appointmentsForPatient.length
    );
  });
  /* -------------------------- - ------------------------- */
  it('7--Displays content correctly after data is successfully fetched', () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <MemoryRouter>
            <Dashboard />
          </MemoryRouter>
        </Provider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  /* -------------------------- - ------------------------- */
  it('8--Opens Appointment form when the edit icon is clicked, fire delete button, opens modal window for confirmation, fire confirm button and trigger delete action', async () => {
    render(
      <>
        <div id="modal" data-testid="modal"></div>
        <Provider store={store}>
          <MemoryRouter>
            <Dashboard />
          </MemoryRouter>
        </Provider>
      </>
    );
    // Check if the modal is rendered
    expect(document.getElementById('modal')).toBeInTheDocument();

    // Simulate clicking the trash icon
    const editIcon = screen.getByTestId('edit-icon');
    fireEvent.click(editIcon);

    expect(screen.getByText('Delete')).toBeInTheDocument();
    expect(screen.getByText('Update')).toBeInTheDocument();
    expect(screen.getByText('Close')).toBeInTheDocument();

    const deleteButton = screen.getByText('Delete');

    fireEvent.click(deleteButton);

    expect(
      screen.getByText('You are about to cancel your recent appointment?')
    ).toBeInTheDocument();
    expect(screen.getByText('Confirm')).toBeInTheDocument();
    expect(screen.getByText('Cancel')).toBeInTheDocument();

    const confirmButton = screen.getByText('Confirm');
    fireEvent.click(confirmButton);

    // Wait for deleteAppointment to be called
    await act(async () => {
      expect(deleteAppointmentMock).toHaveBeenCalledWith(
        '66b0a0c31d2124f7d8aaa65b'
      );
      expect(deleteAppointmentMock).toHaveBeenCalledTimes(1);
    });

    // Additional logging
    console.log(
      'deleteAppointmentMock calls:',
      deleteAppointmentMock.mock.calls
    );

    expect(
      screen.queryByRole('button', { name: 'Confirm' })
    ).not.toBeInTheDocument();
    expect(screen.queryByText('Cancel')).not.toBeInTheDocument();

    expect(screen.queryByText('Delete')).not.toBeInTheDocument();
    expect(screen.queryByText('Update')).not.toBeInTheDocument();
    expect(screen.queryByText('Close')).not.toBeInTheDocument();
  });
  /* -------------------------- - ------------------------- */
  it('9--Opens Appointment form when the edit icon is clicked, fire delete button, opens modal window for confirmation, fire cancel button and close confirmation window', async () => {
    render(
      <>
        <div id="modal" data-testid="modal"></div>
        <Provider store={store}>
          <MemoryRouter>
            <Dashboard />
          </MemoryRouter>
        </Provider>
      </>
    );
    // Check if the modal is rendered
    expect(document.getElementById('modal')).toBeInTheDocument();

    // Simulate clicking the trash icon
    const editIcon = screen.getByTestId('edit-icon');
    fireEvent.click(editIcon);

    expect(screen.getByText('Delete')).toBeInTheDocument();
    expect(screen.getByText('Update')).toBeInTheDocument();
    expect(screen.getByText('Close')).toBeInTheDocument();

    const deleteButton = screen.getByText('Delete');

    fireEvent.click(deleteButton);

    expect(
      screen.getByText('You are about to cancel your recent appointment?')
    ).toBeInTheDocument();
    expect(screen.getByText('Confirm')).toBeInTheDocument();
    expect(screen.getByText('Cancel')).toBeInTheDocument();
    const cancelButton = screen.getByText('Cancel');
    fireEvent.click(cancelButton);

    expect(screen.getByText('Delete')).toBeInTheDocument();
    expect(screen.getByText('Update')).toBeInTheDocument();
    expect(screen.getByText('Close')).toBeInTheDocument();

    expect(screen.queryByText('Confirm')).not.toBeInTheDocument();
    expect(screen.queryByText('Cancel')).not.toBeInTheDocument();
  });
  /* -------------------------- - ------------------------- */
  it('10--open an appointment form by clicking the edit icon, make changes in the inputs, fire the update button, trigger the update action, and finally close the modal window', async () => {
    render(
      <>
        <div id="modal" data-testid="modal"></div>
        <Provider store={store}>
          <MemoryRouter>
            <Dashboard />
          </MemoryRouter>
        </Provider>
      </>
    );
    // Check if the modal is rendered
    expect(document.getElementById('modal')).toBeInTheDocument();

    // Simulate clicking the trash icon
    const editIcon = screen.getByTestId('edit-icon');
    fireEvent.click(editIcon);
   
    //checking modal appointment form presence
    expect(screen.getByText('Appointment Details')).toBeInTheDocument();

    // Simulate changing form inputs
    fireEvent.change(
      screen.getByPlaceholderText('Please write your concerns...'),
      {
        target: { value: 'After update' }, // Example reason change
      }
    );

    // fireEvent.change(screen.getByLabelText('Reason'), {
    //   target: { value: 'Updated Reason' },
    // });

    // Simulate clicking the update button
    const updateButton = screen.getByText('Update');
    fireEvent.click(updateButton);

    await act(async () => {
      expect(updateAppointmentMock).toHaveBeenCalledWith(
        {
          appointmentDateAndTime: '2024-08-16T11:00:00.000Z',
          reason: 'After update',
        },
        '66b0a0c31d2124f7d8aaa65b'
      );
      expect(updateAppointmentMock).toHaveBeenCalledTimes(1);
    });

    // Additional logging
    console.log(
      'updateAppointmentMock calls:',
      updateAppointmentMock.mock.calls
    );

    const fullfilledAction = {
      type: 'appointmentsForPatient/fetchWithIdAndToken/fulfilled',
      payload: {
        appointmentsForPatient: [
          {
            _id: '66b0a0c31d2124f7d8aaa65b',
            doctorId: {
              _id: '665f0f2959c971659af920ca',
              firstName: 'Bowen',
              lastName: 'Chan',
              departmentId: '6660fbd12e3c00fe18cfceef',
            },
            patientId: '6673662fbd42a966b75dec92',
            appointmentDateAndTime: '2024-08-16T11:00:00.000Z',
            reason: 'After update',
            diagnose: null,
            status: null,
            referral: false,
          },
        ],
      },
    };

    await act(async () => {
      store.dispatch(fullfilledAction);
    });

    console.log('Current state:', store.getState().appointmentsForPatient);

    expect(screen.queryByText('Appointment Details')).not.toBeInTheDocument;
    expect(screen.getByText('Latest Appointments')).toBeInTheDocument;

    const cells = screen.getAllByRole('cell');
    expect(cells[4]).toHaveTextContent('After update');

    console.log(store.getState().appointmentsForPatient.status);
    console.log(store.getState().appointmentsForPatient);
  });
  /* -------------------------- - ------------------------- */
  it('11--Renders the error if fetchAppointmentsForPatient returns error', async () => {
    render(
      <Provider store={store}>
        <MemoryRouter>
          <Dashboard />
        </MemoryRouter>
      </Provider>
    );
    await act(async () => {
      store.dispatch({
        type: 'appointmentsForPatient/fetchWithIdAndToken/rejected',
        payload: new Error('Failed to fetch appointments for this patient'),
      });
    });
    expect(
      screen.getByText('Error: Failed to fetch appointments for this patient')
    ).toBeInTheDocument();
  });
});
