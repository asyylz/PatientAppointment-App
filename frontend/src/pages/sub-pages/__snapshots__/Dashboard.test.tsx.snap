// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dasboard 1--Renders correctly before data is fetched 1`] = `
<div
  className="container"
>
  <div
    className="dashboard-left-section"
  >
    <div
      className="dashboard-left-section__wrapper"
    >
      <div
        className="dashboard-left-section__image-wrapper"
      >
        <img
          alt="User profile image"
        />
      </div>
      <p
        className="dashboard-left-section__welcome"
      >
        Welcome 
      </p>
    </div>
  </div>
  <div
    className="dashboard-right-section"
  >
    <div
      className="dashboard-right-section__top"
    >
      <div
        className="dashboard-right-section__top--wrapper dashboard-right-section__top--wrapper--total-visits"
      >
        <p>
          Total Visits: 
        </p>
        <div
          className="icons dashboard-right-section__top--icons--total-visit"
        >
          <svg
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            strokeWidth="0"
            style={
              {
                "color": undefined,
              }
            }
            viewBox="0 0 512 512"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M464 128h-80V80c0-26.5-21.5-48-48-48H176c-26.5 0-48 21.5-48 48v48H48c-26.5 0-48 21.5-48 48v288c0 26.5 21.5 48 48 48h416c26.5 0 48-21.5 48-48V176c0-26.5-21.5-48-48-48zM192 96h128v32H192V96zm160 248c0 4.4-3.6 8-8 8h-56v56c0 4.4-3.6 8-8 8h-48c-4.4 0-8-3.6-8-8v-56h-56c-4.4 0-8-3.6-8-8v-48c0-4.4 3.6-8 8-8h56v-56c0-4.4 3.6-8 8-8h48c4.4 0 8 3.6 8 8v56h56c4.4 0 8 3.6 8 8v48z"
            />
          </svg>
        </div>
      </div>
      <div
        className="dashboard-right-section__top--wrapper dashboard-right-section__top--wrapper--upcoming-visits"
      >
        <p>
          Upcoming Visits: 
        </p>
        <div
          className="icons dashboard-right-section__top--icons--stethoscope"
        >
          <svg
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            strokeWidth="0"
            style={
              {
                "color": undefined,
              }
            }
            viewBox="0 0 512 512"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M447.1 112c-34.2.5-62.3 28.4-63 62.6-.5 24.3 12.5 45.6 32 56.8V344c0 57.3-50.2 104-112 104-60 0-109.2-44.1-111.9-99.2C265 333.8 320 269.2 320 192V36.6c0-11.4-8.1-21.3-19.3-23.5L237.8.5c-13-2.6-25.6 5.8-28.2 18.8L206.4 35c-2.6 13 5.8 25.6 18.8 28.2l30.7 6.1v121.4c0 52.9-42.2 96.7-95.1 97.2-53.4.5-96.9-42.7-96.9-96V69.4l30.7-6.1c13-2.6 21.4-15.2 18.8-28.2l-3.1-15.7C107.7 6.4 95.1-2 82.1.6L19.3 13C8.1 15.3 0 25.1 0 36.6V192c0 77.3 55.1 142 128.1 156.8C130.7 439.2 208.6 512 304 512c97 0 176-75.4 176-168V231.4c19.1-11.1 32-31.7 32-55.4 0-35.7-29.2-64.5-64.9-64zm.9 80c-8.8 0-16-7.2-16-16s7.2-16 16-16 16 7.2 16 16-7.2 16-16 16z"
            />
          </svg>
        </div>
      </div>
      <div
        className="dashboard-right-section__top--wrapper dashboard-right-section__top--wrapper--total-doctors"
      >
        <p>
          Total Doctors : 
          0
        </p>
        <div
          className="icons dashboard-right-section__top--icons--total-doctor"
        >
          <svg
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            strokeWidth="0"
            style={
              {
                "color": undefined,
              }
            }
            viewBox="0 0 448 512"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M224 256A128 128 0 1 0 224 0a128 128 0 1 0 0 256zm-96 55.2C54 332.9 0 401.3 0 482.3C0 498.7 13.3 512 29.7 512H418.3c16.4 0 29.7-13.3 29.7-29.7c0-81-54-149.4-128-171.1V362c27.6 7.1 48 32.2 48 62v40c0 8.8-7.2 16-16 16H336c-8.8 0-16-7.2-16-16s7.2-16 16-16V424c0-17.7-14.3-32-32-32s-32 14.3-32 32v24c8.8 0 16 7.2 16 16s-7.2 16-16 16H256c-8.8 0-16-7.2-16-16V424c0-29.8 20.4-54.9 48-62V304.9c-6-.6-12.1-.9-18.3-.9H178.3c-6.2 0-12.3 .3-18.3 .9v65.4c23.1 6.9 40 28.3 40 53.7c0 30.9-25.1 56-56 56s-56-25.1-56-56c0-25.4 16.9-46.8 40-53.7V311.2zM144 448a24 24 0 1 0 0-48 24 24 0 1 0 0 48z"
            />
          </svg>
        </div>
      </div>
    </div>
    <div
      className="dashboard-right-section__bottom"
    >
      <div
        className="dashboard-right-section__bottom-appointments"
      >
        <div
          className="right-section__bottom-appointments-title"
        >
          <h5>
            Latest Appointments
          </h5>
        </div>
        <hr />
        <table>
          <thead>
            <tr>
              <th>
                No
              </th>
              <th>
                Doctor Name
              </th>
              <th>
                Date
              </th>
              <th>
                Time
              </th>
              <th>
                Concerns
              </th>
              <th>
                Diagnose
              </th>
              <th>
                Actions
              </th>
            </tr>
          </thead>
          <tbody
            data-testid="appointments-patient"
          >
            <tr
              className="gap--line"
            />
          </tbody>
        </table>
        <div
          className="pagination__wrapper"
        >
          <div
            className="pagination__icon--wrapper"
            onClick={[Function]}
          >
            <svg
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              strokeWidth="0"
              style={
                {
                  "color": undefined,
                }
              }
              viewBox="0 0 24 24"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <polyline
                fill="none"
                points="7 2 17 12 7 22"
                strokeWidth="2"
                transform="matrix(-1 0 0 1 24 0)"
              />
            </svg>
            <p>
              Prev
            </p>
          </div>
          <p
            className="pagination__page-number"
          >
            Page: 
            1
          </p>
          <div
            className="pagination__icon--wrapper"
            onClick={[Function]}
          >
            <p>
              Next
            </p>
            <svg
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              strokeWidth="0"
              style={
                {
                  "color": undefined,
                }
              }
              viewBox="0 0 24 24"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <polyline
                fill="none"
                points="7 2 17 12 7 22"
                strokeWidth="2"
              />
            </svg>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Dasboard 7--Displays content correctly after data is successfully fetched 1`] = `
<div
  className="container"
>
  <div
    className="dashboard-left-section"
  >
    <div
      className="dashboard-left-section__wrapper"
    >
      <div
        className="dashboard-left-section__image-wrapper"
      >
        <img
          alt="User profile image"
        />
      </div>
      <p
        className="dashboard-left-section__welcome"
      >
        Welcome 
      </p>
    </div>
  </div>
  <div
    className="dashboard-right-section"
  >
    <div
      className="dashboard-right-section__top"
    >
      <div
        className="dashboard-right-section__top--wrapper dashboard-right-section__top--wrapper--total-visits"
      >
        <p>
          Total Visits: 
        </p>
        <div
          className="icons dashboard-right-section__top--icons--total-visit"
        >
          <svg
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            strokeWidth="0"
            style={
              {
                "color": undefined,
              }
            }
            viewBox="0 0 512 512"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M464 128h-80V80c0-26.5-21.5-48-48-48H176c-26.5 0-48 21.5-48 48v48H48c-26.5 0-48 21.5-48 48v288c0 26.5 21.5 48 48 48h416c26.5 0 48-21.5 48-48V176c0-26.5-21.5-48-48-48zM192 96h128v32H192V96zm160 248c0 4.4-3.6 8-8 8h-56v56c0 4.4-3.6 8-8 8h-48c-4.4 0-8-3.6-8-8v-56h-56c-4.4 0-8-3.6-8-8v-48c0-4.4 3.6-8 8-8h56v-56c0-4.4 3.6-8 8-8h48c4.4 0 8 3.6 8 8v56h56c4.4 0 8 3.6 8 8v48z"
            />
          </svg>
        </div>
      </div>
      <div
        className="dashboard-right-section__top--wrapper dashboard-right-section__top--wrapper--upcoming-visits"
      >
        <p>
          Upcoming Visits: 
        </p>
        <div
          className="icons dashboard-right-section__top--icons--stethoscope"
        >
          <svg
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            strokeWidth="0"
            style={
              {
                "color": undefined,
              }
            }
            viewBox="0 0 512 512"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M447.1 112c-34.2.5-62.3 28.4-63 62.6-.5 24.3 12.5 45.6 32 56.8V344c0 57.3-50.2 104-112 104-60 0-109.2-44.1-111.9-99.2C265 333.8 320 269.2 320 192V36.6c0-11.4-8.1-21.3-19.3-23.5L237.8.5c-13-2.6-25.6 5.8-28.2 18.8L206.4 35c-2.6 13 5.8 25.6 18.8 28.2l30.7 6.1v121.4c0 52.9-42.2 96.7-95.1 97.2-53.4.5-96.9-42.7-96.9-96V69.4l30.7-6.1c13-2.6 21.4-15.2 18.8-28.2l-3.1-15.7C107.7 6.4 95.1-2 82.1.6L19.3 13C8.1 15.3 0 25.1 0 36.6V192c0 77.3 55.1 142 128.1 156.8C130.7 439.2 208.6 512 304 512c97 0 176-75.4 176-168V231.4c19.1-11.1 32-31.7 32-55.4 0-35.7-29.2-64.5-64.9-64zm.9 80c-8.8 0-16-7.2-16-16s7.2-16 16-16 16 7.2 16 16-7.2 16-16 16z"
            />
          </svg>
        </div>
      </div>
      <div
        className="dashboard-right-section__top--wrapper dashboard-right-section__top--wrapper--total-doctors"
      >
        <p>
          Total Doctors : 
          1
        </p>
        <div
          className="icons dashboard-right-section__top--icons--total-doctor"
        >
          <svg
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            strokeWidth="0"
            style={
              {
                "color": undefined,
              }
            }
            viewBox="0 0 448 512"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M224 256A128 128 0 1 0 224 0a128 128 0 1 0 0 256zm-96 55.2C54 332.9 0 401.3 0 482.3C0 498.7 13.3 512 29.7 512H418.3c16.4 0 29.7-13.3 29.7-29.7c0-81-54-149.4-128-171.1V362c27.6 7.1 48 32.2 48 62v40c0 8.8-7.2 16-16 16H336c-8.8 0-16-7.2-16-16s7.2-16 16-16V424c0-17.7-14.3-32-32-32s-32 14.3-32 32v24c8.8 0 16 7.2 16 16s-7.2 16-16 16H256c-8.8 0-16-7.2-16-16V424c0-29.8 20.4-54.9 48-62V304.9c-6-.6-12.1-.9-18.3-.9H178.3c-6.2 0-12.3 .3-18.3 .9v65.4c23.1 6.9 40 28.3 40 53.7c0 30.9-25.1 56-56 56s-56-25.1-56-56c0-25.4 16.9-46.8 40-53.7V311.2zM144 448a24 24 0 1 0 0-48 24 24 0 1 0 0 48z"
            />
          </svg>
        </div>
      </div>
    </div>
    <div
      className="dashboard-right-section__bottom"
    >
      <div
        className="dashboard-right-section__bottom-appointments"
      >
        <div
          className="right-section__bottom-appointments-title"
        >
          <h5>
            Latest Appointments
          </h5>
        </div>
        <hr />
        <table>
          <thead>
            <tr>
              <th>
                No
              </th>
              <th>
                Doctor Name
              </th>
              <th>
                Date
              </th>
              <th>
                Time
              </th>
              <th>
                Concerns
              </th>
              <th>
                Diagnose
              </th>
              <th>
                Actions
              </th>
            </tr>
          </thead>
          <tbody
            data-testid="appointments-patient"
          >
            <tr
              className="gap--line"
            />
            <tr
              className="row"
            >
              <td>
                1
                .
              </td>
              <td>
                Dr. Bowen Chan
              </td>
              <td>
                16/08/2024 Friday
              </td>
              <td>
                11:00
              </td>
              <td>
                Before update
              </td>
              <td />
              <td>
                <svg
                  className="icons icons--edit"
                  data-testid="edit-icon"
                  fill="currentColor"
                  height="1em"
                  onClick={[Function]}
                  stroke="currentColor"
                  strokeWidth="0"
                  style={
                    {
                      "color": undefined,
                    }
                  }
                  viewBox="0 0 576 512"
                  width="1em"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M402.6 83.2l90.2 90.2c3.8 3.8 3.8 10 0 13.8L274.4 405.6l-92.8 10.3c-12.4 1.4-22.9-9.1-21.5-21.5l10.3-92.8L388.8 83.2c3.8-3.8 10-3.8 13.8 0zm162-22.9l-48.8-48.8c-15.2-15.2-39.9-15.2-55.2 0l-35.4 35.4c-3.8 3.8-3.8 10 0 13.8l90.2 90.2c3.8 3.8 10 3.8 13.8 0l35.4-35.4c15.2-15.3 15.2-40 0-55.2zM384 346.2V448H64V128h229.8c3.2 0 6.2-1.3 8.5-3.5l40-40c7.6-7.6 2.2-20.5-8.5-20.5H48C21.5 64 0 85.5 0 112v352c0 26.5 21.5 48 48 48h352c26.5 0 48-21.5 48-48V306.2c0-10.7-12.9-16-20.5-8.5l-40 40c-2.2 2.3-3.5 5.3-3.5 8.5z"
                  />
                </svg>
              </td>
            </tr>
            <tr
              className="gap--line"
            />
          </tbody>
        </table>
        <div
          className="pagination__wrapper"
        >
          <div
            className="pagination__icon--wrapper"
            onClick={[Function]}
          >
            <svg
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              strokeWidth="0"
              style={
                {
                  "color": undefined,
                }
              }
              viewBox="0 0 24 24"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <polyline
                fill="none"
                points="7 2 17 12 7 22"
                strokeWidth="2"
                transform="matrix(-1 0 0 1 24 0)"
              />
            </svg>
            <p>
              Prev
            </p>
          </div>
          <p
            className="pagination__page-number"
          >
            Page: 
            1
          </p>
          <div
            className="pagination__icon--wrapper passive"
            onClick={[Function]}
          >
            <p>
              Next
            </p>
            <svg
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              strokeWidth="0"
              style={
                {
                  "color": undefined,
                }
              }
              viewBox="0 0 24 24"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <polyline
                fill="none"
                points="7 2 17 12 7 22"
                strokeWidth="2"
              />
            </svg>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
